secret-engine/src/main/java/com/sequenceiq/cloudbreak/service/secret/service/rotation/context/VaultRotationContext.java

```diff
@@ -3,19 +3,19 @@
 import java.util.Map;
 
 import com.sequenceiq.cloudbreak.rotation.secret.RotationContext;
-import com.sequenceiq.cloudbreak.service.secret.domain.Secret;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretGenerator;
 
 public class VaultRotationContext extends RotationContext {
 
-    private final Map<Secret, String> newSecretMap;
+    private final Map<String, SecretGenerator> secretUpdateSupplierMap;
```

Comment:
The SecretGenerator might not be serializable to the flow log. I had an implementation where this map is a:

```java
Map<String, Class<? extends SecretGenerator>>
```

All secret generators can be a component injected into `VaultRotationExecutor` and be looked up based on the class.